src/BloomBrowserUI/bookEdit/js/bloomEditing.js

```diff
@@ -1141,6 +1143,51 @@ $.fn.hasAttr = function (name) {
     return (typeof attr !== 'undefined' && attr !== false);
 };
 
+// Some custom templates have image containers embedded in bloom-editable divs, so that the text can wrap
+// around the picture. The problems is that the user can do ctrl+a, del to start over on the text, and 
+// inadvertantly remove the embedded images. So we introduced the "bloom-preventRemoval" class, and this
+// tries to safeguard element bearing that class.
+function PreventRemovalOfSomeElements(container) {
+    
+    /* this approach showed promise, but only the first time you do ctrl+all, DEL. After the undo, the bindins were not redone. 
+    $(container).find(".bloom-preventRemoval").bind("DOMNodeRemoved", function (e) {
+        alert("Removed: " + e.target.nodeName);
+        //this threw a NS_ERROR but I don't know why
+        //document.execCommand('undo', false, null);
+    });
+
+    the problem with this one is the event was raised when we weren't actually deleting it
+    $(container).bind("DOMNodeRemoved", function (e) {
+        if ($(e.target).hasClass('bloom-preventRemoval')) {
+            alert("Removed: " + e.target.nodeName);
+            document.execCommand('undo', false, null);
+        }
+        //this threw a NS_ERROR but I don't know why
+        //document.execCommand('undo', false, null);
+    });
+    */
+    //This works for the actual case that I was trying to prevent, where the user
+    //does a ctrl+all and then delete. But notice that it's not logically sound;
+    //so long as at least *one* bloom-preventRemoval child is still there, this
+    //wouldn't trigger.
+    //Note, the is event is *not* fired on the element itself in the ctrl+all, del scenario, hence
+    //the need to go up to the parent editable and attach the even their.
+    $(container).find(".bloom-preventRemoval").closest(".bloom-editable").on("input", function (e) {
+        if ($(this).find(".bloom-preventRemoval").length == 0) {
+            document.execCommand('undo');
+        }
```

Comment:
Could this be improved by counting the original preventRemoval elements, and preserving this as part of the context of the function, so it undoes if the number changes?
Something like ..closest(".bloom-editable").each(function(parent) {
  var count = parent.find(".bloom-preventRemoval").length;
  parent.on("input", function(e) {
    if ($(this).find(".bloom-preventRemoval").length != count)...